### ISSUE TITLE
kedro pipeline create <package_name> stop working in 0.16.6 version

### ISSUE THREAD

**Issue by hec10r** [quantumblacklabs/kedro#643]
## Description

I am trying to create a modular pipeline following [this](https://kedro.readthedocs.io/en/0.16.6/06_nodes_and_pipelines/02_pipelines.html#how-do-i-create-modular-pipelines) and when I run `kedro pipeline create my_package_name` I am getting this error: 
```
Traceback (most recent call last):
  File "/Users/hector_lopez/anaconda3/envs/acube/bin/kedro", line 8, in <module>
    sys.exit(main())
  File "/Users/hector_lopez/anaconda3/envs/acube/lib/python3.7/site-packages/kedro/framework/cli/cli.py", line 725, in main
    cli_collection()
  File "/Users/hector_lopez/anaconda3/envs/acube/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/hector_lopez/anaconda3/envs/acube/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/hector_lopez/anaconda3/envs/acube/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/hector_lopez/anaconda3/envs/acube/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/hector_lopez/anaconda3/envs/acube/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/hector_lopez/anaconda3/envs/acube/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/hector_lopez/anaconda3/envs/acube/lib/python3.7/site-packages/kedro/framework/cli/pipeline.py", line 104, in create_pipeline
    package_dir = _get_project_package_dir(context)
  File "/Users/hector_lopez/anaconda3/envs/acube/lib/python3.7/site-packages/kedro/framework/cli/pipeline.py", line 503, in _get_project_package_dir
    project_package = import_module(context.package_name)
  File "/Users/hector_lopez/anaconda3/envs/acube/lib/python3.7/site-packages/kedro/framework/context/context.py", line 298, in package_name
    return self.static_data["package_name"]
KeyError: 'package_name'
``` 

## Context
This works in `kedro 0.16.4`. If I downgrade my kedro version it will do the job. Is there any reason for this change?

## Steps to Reproduce
1. Install kedro 0.16.6
2. Create a kedro project
3. Run `kedro pipeline create <package_name>`

## Expected Result
Not having an error

## Actual Result
An error

## Your Environment
Include as many relevant details about the environment in which you experienced the bug:

* Kedro version used: **kedro, version 0.16.6**
* Python version used: **Python 3.7.0**
* Operating system and version: **macOS Caralina 10.15.7**

**Comment by hec10r** [comment]
It also doesn't work in `0.16.5`

**Comment by hec10r** [comment]
The problem seems to be that the project was created using an older version of kedro.

I tried creating a project using 0.16.6 and running the command and it works, but if I downgrade to 0.16.4 and try to run the same command in this new project I get this:

```
TypeError: Can't instantiate abstract class ProjectContext with abstract methods project_name, project_version
Run with --verbose to see the full exception
Error: Unable to load Kedro context with environment `None`. Make sure it exists in the project configuration.
Error: Can't instantiate abstract class ProjectContext with abstract methods project_name, project_version
```

In conclusion: Do not upgrade kedro version in the middle of a project

**Comment by lorenabalan** [comment]
Hi @hec10r,  sadly I've not been able to reproduce the issue. I've generated a project with 0.16.4, and `kedro pipeline create` worked with all of 0.16.4, 0.16.5, 0.16.6 versions. `kedro pipeline create` also works in a project generated with 0.16.6, with 0.16.6 installed. 

What I think is causing more of the confusion is the way we load starters, if that's what you've used in testing - then I got the same error you mentioned in the comment above. 

```
kedro new --starter pandas-iris
```

This now maps the version of the starter with the version of kedro installed locally. In older versions though, this was fetching the latest master version of the starter, which wasn't guaranteed to work - particularly with older versions of kedro. To make it more robust I recommend running 

```
kedro new --starter pandas-iris --checkout 0.16.4
```
